rotkehlchen/db/snapshots.py

```diff
@@ -223,41 +223,57 @@ def _export(
 
         return True, ''
 
-    def import_snapshot(
-        self,
-        balances_snapshot_file: Path,
-        location_data_snapshot_file: Path,
+    @staticmethod
+    def validate_import_data(
+        balances_data: List[Dict[str, str]],
```

Comment:
I would split validation into 2 parts. First - for creating data structures form json and second - for validating timestamps and other properties. I would do this since you need to validate data at various steps.